mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -879,6 +879,13 @@ private MarcBibCollection toMarcBibCollection(QueryResult result) {
     return SnapshotDaoUtil.delete(getQueryExecutor(tenantId), snapshotId);
   }
 
+  @Override
+  public Future<Boolean> deleteRecordById(String recordId, String tenantId) {
+    return getQueryExecutor(tenantId).execute(dsl -> dsl.deleteFrom(RECORDS_LB)
+        .where(RECORDS_LB.ID.eq(UUID.fromString(recordId))))
+      .map(res -> res == 1);
+  }
```

Comment:
It was decided to use a soft-delete approach for deleting record in SRS. It means that the record should have state = DELETED in the database.